### PR TITLE
virtio_serial: offline_migration with virtio serial devices

### PR DESCRIPTION
**Pull request by nanliu-r**
ID: 1755680

Signed-off-by: Nana Liu <nanliu@redhat.com>

### CODE REVIEW

**Comment by nanliu-r** [comment]
Test Pass:

(100/100) repeat100.Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.offline_migration:  PASS (72.14 s)
RESULTS    : PASS 100 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

 (100/100) repeat100.Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.offline_migration:  PASS (142.83 s)
RESULTS    : PASS 100 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

**Comment by nanliu-r** [comment]
@yanan-fu Please help to review this, thanks a lot.

**Comment by yanan-fu** [comment]
From the test case, the test scenario is transfer data during `migrate via a compressed file`. 
what you implemented is migrate with tcp, please check again!

File: qemu/tests/cfg/virtio_serial_file_transfer.cfg [hunk 1]
```diff
@@ -43,6 +43,15 @@
             type = virtio_serial_file_transfer_max_ports
             start_vm = no
             virtio_serial_ports = 30
+        - offline_migration:
+            only unix_socket
+            type = virtio_serial_file_transfer_offline_migrate
+            serials += " vs2"
+            serial_type_vs1 = virtserialport
```
Comment (yanan-fu) [inline qemu/tests/cfg/virtio_serial_file_transfer.cfg]: vs1 already be added with L6~L7.  no serial_type for the new serial vs2.
And could you explain the reason for adding vs2 ?

Comment (nanliu-r) [reply qemu/tests/cfg/virtio_serial_file_transfer.cfg]: sorry, it should be `serial_type_vs2` and already updated.

Comment (yanan-fu) [reply qemu/tests/cfg/virtio_serial_file_transfer.cfg]: @nanliu-r could you explain the reason of adding vs2 but do not use it?

Comment (nanliu-r) [reply qemu/tests/cfg/virtio_serial_file_transfer.cfg]: > @nanliu-r could you explain the reason of adding vs2 but do not use it?

To make the test more reasonable I will add a test of vs2, thanks.

**Comment by nanliu-r** [comment]
Test results:

 Host_RHEL.m7.u9.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.8.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.offline_migration: PASS (103.24 s)
Host_RHEL.m7.u9.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.offline_migration: PASS (324.67 s)

File: qemu/tests/cfg/virtio_serial_file_transfer.cfg (hunk 1 above)
Comment (yanan-fu) [reply qemu/tests/cfg/virtio_serial_file_transfer.cfg]: @nanliu-r As we discussed, test both vs1 and vs2 before migrate and after migrate, and update polarion case too.

File: qemu/tests/cfg/virtio_serial_file_transfer.cfg [hunk 2]
```diff
@@ -43,6 +43,15 @@
             type = virtio_serial_file_transfer_max_ports
             start_vm = no
             virtio_serial_ports = 30
+        - offline_migration:
+            only unix_socket
+            type = virtio_serial_file_transfer_offline_migrate
+            serials += " vs2"
+            serial_type_vs2 = virtserialport
+            file_transfer_serial_port = vs1
```
Comment (yanan-fu) [inline qemu/tests/cfg/virtio_serial_file_transfer.cfg]: If test two ports,  remove this definition, traversal serials start with 'virt' in script is ok.

Comment (nanliu-r) [reply qemu/tests/cfg/virtio_serial_file_transfer.cfg]: > If test two ports, remove this definition, traversal serials start with 'virt' in script is ok.

update and thanks

File: qemu/tests/cfg/virtio_serial_file_transfer.cfg (hunk 1 above)
Comment (nanliu-r) [reply qemu/tests/cfg/virtio_serial_file_transfer.cfg]: thanks and update.

**Comment by nanliu-r** [comment]
Latest test results:

 (1/2) Host_RHEL.m7.u9.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.8.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.offline_migration: PASS (142.05 s)
 (2/2) Host_RHEL.m7.u9.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.offline_migration: PASS (254.96 s)

**Comment by yanan-fu** [comment]
depends on https://github.com/avocado-framework/avocado-vt/pull/2406

**Pull request merged**